content/productivity/pink-it/plugin.js

```diff
@@ -0,0 +1,529 @@
+	table.appendChild(tr);
+}
+
+function addAsTd(tr, text, width=null, color=null, center=true) {
+	var td = document.createElement('td');
+	td.style.border = "1px solid white";
+	td.innerHTML = text;
+	if (width) td.width = width+"px";
+	if (color) td.style.color = color;
+	if (center) td.style["text-align"] = "center";
+	tr.appendChild(td);
+	return td;
+}
+
+function intToStrHexColor(num) {
+	var color = num.toString(16).toUpperCase();
+	if (color.length < 2) color = "0"+color;
+	return color;
+}
+function percentToStrHexColor(percent) {
+	return intToStrHexColor(parseInt(percent*255));
+}
+function getPlanetRingRadius(planet, multiplier=1.3, min=minDrawPlanetRadius) {
+	const viewport = ui.getViewport();
+	let radius = viewport.worldToCanvasDist(ui.getRadiusOfPlanetLevel(planet.planetLevel));
+	radius *= multiplier;
+	if (radius < min) radius = min;
+	return radius;
+}
+function drawRingAnimationOnPlanet(ctx, planet, level, color, reverse=false) {
+	const viewport = ui.getViewport();
+	const { x: planetX, y: planetY } = viewport.worldToCanvasCoords(planet.location.coords);
+
+	let radius = getPlanetRingRadius(planet);
+
+	let animationDuration = 1500;
+	animationDuration *= (1 - level*0.1);
+	
+	var millis = Date.now();
+
+	for (var i=0; i<=level; ++i) {
+		drawRingAnimation(ctx, planetX, planetY, radius, millis, animationDuration, color, reverse);
+		millis += animationDuration / (level+1);
+	}
+}
+function drawRingAnimation(ctx, x, y, radius, millis, animationDuration, color, reverse=false) {
+	let time = (millis % animationDuration) / animationDuration;
+	if (reverse) time = 1 - time;
+	const thickness = 1 - time*0.2;
+	let opacity = 1 - time;
+	if (opacity > 0.6) opacity *= 1-(opacity-0.6)/0.4;
+	opacity = percentToStrHexColor(opacity);
+	radius += radius * time * 1.1;
+	drawRing2(ctx, x, y, radius, thickness, color+opacity);
+}
+function drawRing2(ctx, x, y, radius, thickness=0.8, color) {
+	ctx.beginPath();
+	ctx.strokeStyle = color;
+	ctx.fillStyle = color;
+	ctx.arc(x, y, radius, 0, PI2, false);
+	ctx.arc(x, y, radius * thickness, 0, PI2, true);
+	ctx.fill();
+}
+
+function drawPinkZone(ctx, coords, color="#FFF") {
+	const viewport = ui.getViewport();
+	const { x, y } = viewport.worldToCanvasCoords(coords);
+	const radius = viewport.worldToCanvasDist(pinkCircleRadius);
+	ctx.beginPath();
+	ctx.fillStyle = color;
+	ctx.arc(x, y, radius, 0, PI2, false);
+	ctx.fill();
+}
+
+function drawHlPinkZone(ctx) {
+	if (hlPinkZoneCounter <= 0) return;
+	let selectedPlanet = ui.getSelectedPlanet();
+	if (!selectedPlanet) return;
+	let color = "hsla(0, 100%, 40%, "+(hlPinkZoneCounter/hlPinkZoneCounterMax*0.4)+")";
+	drawPinkZone(ctx, selectedPlanet.location.coords, color);
+	hlPinkZoneCounter--;
+}
+
+function drawListHoveringPZ(ctx) {
+	if (!pzHoveringInList) return;
+	let selectedPlanet = ui.getSelectedPlanet();
+//	if (selectedPlanet && pzHoveringInList.locationId === selectedPlanet.locationId) return;
+	let color = "hsla(300, 100%, 40%, 0.7)";
```

Comment:
```suggestion
	const color = "hsla(300, 100%, 40%, 0.7)";
```

immutable variable here.